### PR TITLE
ReactEditText extends AppCompatEditText

### PR DESCRIPTION
**Pull request by dulmandakh**
## Summary

Google recommends to extend AppCompat widgets, and Android Studio suggests the change. This PR changes ReactEditText to extend AppCompatEditText.

## Changelog

[Android] [Changed] - ReactEditText extends AppCompatEditText

## Test Plan

CI is green

### CODE REVIEW

**Comment by mdvacca** [comment]
@dulmandakh This PR failed to land because there are internal components that inherit from ReactTextInputManager that now require to import the androidx.appcompat.widget.AppCompatEditText dependency. 
I will modified that when I have some time, and I will try to re-land

**Comment by dulmandakh** [comment]
@mdvacca you can use exported_deps

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java [hunk 1]
```diff
@@ -30,10 +30,10 @@
 import android.view.inputmethod.EditorInfo;
 import android.view.inputmethod.InputConnection;
 import android.view.inputmethod.InputMethodManager;
-import android.widget.EditText;
 import androidx.annotation.Nullable;
 import androidx.core.view.AccessibilityDelegateCompat;
 import androidx.core.view.ViewCompat;
+import androidx.appcompat.widget.AppCompatEditText;
```
Comment (friederbluemle) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java]: Minor: Keep in alphabetical order please (should be inserted above `androidx.core`)

**Comment by dulmandakh** [comment]
@hramos @mdvacca please review and merge

**Pull request closed without merging**